components/ws-manager/pkg/manager/pod_controller.go

```diff
@@ -34,8 +36,20 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 	err := r.Client.Get(context.Background(), req.NamespacedName, &pod)
 	if errors.IsNotFound(err) {
 		// pod is gone - that's ok
+		if pod, ok := r.Pods[req.NamespacedName]; !ok {
```

Comment:
`if pod, ok := r.Pods[req.NamespacedName]; !ok {`

The `!ok` means pod does not exist in the Pods map.
So, I wonder how it is possible that the pod can add to the Deleted event pool?

Reply:
Can you please reverify these cases https://github.com/gitpod-io/gitpod/pull/14321#issuecomment-1310074302? Thank you.

Reply:
Done